### ISSUE TITLE
ALSA audio device slave is hardcoded in the release

### ISSUE THREAD

**Issue by ivan-ristovic** [matf-pp/2023_Pesak#12]
```
ALSA lib pcm_dmix.c:999:(snd_pcm_dmix_open) unable to open slave
panic: ALSA: Couldn't open audio device: Device or resource busy

goroutine 1 [running, locked to thread]:
main/musicPack.OpenAudio(...)
        /home/njanja/Desktop/2023_Pesak/musicPack/musicLib.go:19
main.main()
        /home/njanja/Desktop/2023_Pesak/main.go:52 +0x845
```

It seems like the paths are hardcoded in this [linux release](https://github.com/matf-pp/2023_Pesak/releases/tag/v1.3.0-stable) (making it impossible to be opened).

**Comment by sajberk** [comment]
I don't think this issue has anything to do with hardcoded paths. I believe the absolute paths appear due to the way Cgo and gcc handle static compilation of C code, which is necessary for the SDL libraries written in C. There is an explanation of a similar issue in [this](https://stackoverflow.com/questions/45856955/how-to-tell-cmake-to-use-relative-paths?noredirect=1&lq=1) thread. In the end, absolute paths showing in the error log is actually expected behavior, as they only show up when there is an error that would also have happened otherwise. We have encountered it many times already, and each time we've "solved" it by fixing the underlying bug that caused the error to surface in the first place.

That said, the function that returned the error is SDL's [Mix_OpenAudio](https://wiki.libsdl.org/SDL2_mixer/Mix_OpenAudio), which doesn't depend on a specific path anyway, and just opens the default audio device for playback. Furthermore, many people have been able to run that specific Linux release, which makes me think that the issue is definitely on your end. Can you double check that your audio is set up properly? Could it be that you tested this on a VM without proper drivers or audio devices connected? 

I haven't been able to reproduce the bug, and I don't think I can do much if the user does not have an audio device available anyway. The only fix I can propose right now is to not play music instead of crashing if SDL fails to find an available device. Would that suffice?

I do not completely understand the cause of your issue, so I would be thankful for any help in understanding and resolving it.

**Comment by ivan-ristovic** [comment]
Actually I realized in the meantime that opening the device fails if the device is already in use by another program. I had mpd running in the background and this error appears 100% of the time. When not having any other programs that use the audio device, it works fine. You can try to run the program while having background music and see if the error appears.

So, the core of the issue here (I believe) is that one ALSA device (~~iirc~~ edit: [SO link that backs this theory up](https://askubuntu.com/questions/581128/what-is-the-relation-between-alsa-and-pulseaudio-sound-architecture)) cannot be used by multiple programs. Generally programs solve this problem by using a software proxy (e.g., PulseAudio). Another (kind of annoying) side effect is that users can't have other sounds playing while Pesak is running.

You can decide if this is a "feature" or not. In my opinion this is really annoying and I think we need at least a better error message. Another option is not to fail in such cases and simply not have music playing if opening the device fails (i.e., not panic).

**Comment by sajberk** [comment]
I imagined that something like that has happened to you, so I tried playing Pesak with other programs playing sound in the background, and it worked as expected. Even multiple instances of Pesak were able to play sound at the same time, so I can't reproduce the bug. 

PulseAudio is the default audio driver used by SDL2 ([source](https://wiki.libsdl.org/SDL2/FAQUsingSDL)), so I don't really think I can do much here other than provide a better error message. I'd gladly fix the bug if I knew how.
